### DISCUSSION TITLE
Executing SDLs programmatically

### DISCUSSION THREAD

**Discussion by rajrahul** [geldata/gel#3490] [Q&A] [answered]
I see that to the dbschema/default.esdl file needs to be maintained in order to create the required tables. How is it possible to achieve this programmatically? Currently we have a server application that maintains the tables and manages the migrations.  How don't see any documented approach to this.

**Comment by elprans** [comment, marked as answer]
The CLI actually just uses the [migration DDL](https://www.edgedb.com/docs/reference/ddl/migrations).  You can emulate what the CLI does by using the migration commands, including the interactive part.
